tests/components/accuweather/test_sensor.py

```diff
@@ -0,0 +1,589 @@
+        == CONCENTRATION_PARTS_PER_CUBIC_METER
+    )
+    assert state.attributes.get("level") == "Low"
+    assert state.attributes.get(ATTR_ICON) == "mdi:tree-outline"
+
+    entry = registry.async_get("sensor.home_tree_pollen_0d")
+    assert entry
+    assert entry.unique_id == "0123456-tree-0"
+
+    state = hass.states.get("sensor.home_wind_gust_day_0d")
+    assert state
+    assert state.state == "29.6"
+    assert state.attributes.get(ATTR_ATTRIBUTION) == ATTRIBUTION
+    assert state.attributes.get(ATTR_UNIT_OF_MEASUREMENT) == SPEED_KILOMETERS_PER_HOUR
+    assert state.attributes.get("direction") == "S"
+    assert state.attributes.get(ATTR_ICON) == "mdi:weather-windy"
+
+    entry = registry.async_get("sensor.home_wind_gust_day_0d")
+    assert entry
+    assert entry.unique_id == "0123456-windgustday-0"
+
+    state = hass.states.get("sensor.home_wind_gust_night_0d")
+    assert state
+    assert state.state == "18.5"
+    assert state.attributes.get(ATTR_ATTRIBUTION) == ATTRIBUTION
+    assert state.attributes.get(ATTR_UNIT_OF_MEASUREMENT) == SPEED_KILOMETERS_PER_HOUR
+    assert state.attributes.get("direction") == "WSW"
+    assert state.attributes.get(ATTR_ICON) == "mdi:weather-windy"
+
+    entry = registry.async_get("sensor.home_wind_gust_night_0d")
+    assert entry
+    assert entry.unique_id == "0123456-windgustnight-0"
+
+
+async def test_availability(hass):
+    """Ensure that we mark the entities unavailable correctly when service is offline."""
+    await init_integration(hass)
+
+    state = hass.states.get("sensor.home_cloud_ceiling")
+    assert state
+    assert state.state != STATE_UNAVAILABLE
+    assert state.state == "3200"
+
+    future = utcnow() + timedelta(minutes=60)
+    with patch(
+        "accuweather.AccuWeather.async_get_current_conditions",
+        side_effect=ConnectionError(),
+    ):
+        async_fire_time_changed(hass, future)
+        await hass.async_block_till_done()
+
+        state = hass.states.get("sensor.home_cloud_ceiling")
+        assert state
+        assert state.state == STATE_UNAVAILABLE
+
+    future = utcnow() + timedelta(minutes=120)
+    with patch(
+        "accuweather.AccuWeather.async_get_current_conditions",
+        return_value=json.loads(
+            load_fixture("accuweather/current_conditions_data.json")
+        ),
+    ):
+        async_fire_time_changed(hass, future)
+        await hass.async_block_till_done()
+
+        state = hass.states.get("sensor.home_cloud_ceiling")
+        assert state
+        assert state.state != STATE_UNAVAILABLE
+        assert state.state == "3200"
+
+
+async def test_manual_update_entity(hass):
+    """Test manual update entity via service homeasasistant/update_entity."""
+    await init_integration(hass)
+
+    await async_setup_component(hass, "homeassistant", {})
+    with patch(
+        "homeassistant.components.accuweather.AccuWeatherDataUpdateCoordinator._async_update_data"
```

Comment:
Please patch the library instead and assert the call to that. If needed we can patch the coordinator debounce time constant.